scripts/o3de/o3de/repo.py

```diff
@@ -82,6 +82,9 @@ def download_object_manifests(repo_data):
                 return download_file_result
     return 0
 
+def get_repo_schema_version(repo_data: dict):
+    return repo_data.get("$schemaVersion", "1.0.0")
```

Comment:
First it seems that the schema version should be a constant for the current version
i.e
```python
REPO_CURRENT_SCHEMA_VERSION = "1.0.0"
```

Next the current repo.json didn't have a version before which for consistency with our version numbering scheme we use in other O3DE manifest should be [0.0.0](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/PythonBindings.cpp#L402-L412)

Reply:
given that the current version could be upgraded at any time, Im thinking the constant names should just spell out their version ID instead.

something like:
```
REPO_SCHEMA_VERSION_0_0_0 = "0.0.0"
```

Reply:
Regarding the default schema version, I'm thinking it should be "implicit" schema version, not "current"